### PR TITLE
Show User Events on dedicated tab on Client-/User-Details

### LINKED ISSUE #33569: Show User Events on dedicated tab on Client-/User-Details
### Description

Partially part of https://github.com/keycloak/keycloak/issues/29113

To quickly be able to investigate the user events of a client or user, dedicated tabs on the user and client details should be added:

Client Details:
image

User Details:
image

The visible events are prefiltered for the corresponding client-/user but allow to add additional filters. These tabs are only visible for users with view-events permission.

### Discussion

_No response_

### Motivation

Better UX when trying to find user or client events

### Details

I will create a PR.

### PR DESCRIPTION
**Pull request by antikalk**
Fixes #33569

Client Details:
image

User Details:
image

Unfortunately I missed the 26.0.0 release :( ... any chance of getting this backported to KC26?

I had to split the UserEvents tab content from the EventsSection in order to reuse it for the client/user details tab.

### CODE REVIEW

File: js/apps/admin-ui/src/events/UserEvents.tsx [hunk 1]
```diff
@@ -0,0 +1,476 @@
...
+        >
+          {event.userId}
+        </Link>
+      )}
+      {!event.userId && t("noUserDetails")}
+    </>
+  );
+};
+
+type UserEventsProps = {
```
Comment (antikalk) [inline js/apps/admin-ui/src/events/UserEvents.tsx:114]: These Props are new, to pass a user and/or client to filter on.

File: js/apps/admin-ui/src/events/UserEvents.tsx [hunk 2]
```diff
@@ -0,0 +1,476 @@
...
+                        ))}
+                      </KeycloakSelect>
+                    )}
+                  />
+                </FormGroup>
+                <TextControl
+                  name="client"
+                  label={t("client")}
+                  data-testid="client-searchField"
+                  isDisabled={!!client}
```
Comment (antikalk) [inline js/apps/admin-ui/src/events/UserEvents.tsx:317]: Disable input if a client has been passed to the component.

File: js/apps/admin-ui/src/events/UserEvents.tsx [hunk 3]
```diff
@@ -0,0 +1,476 @@
...
+                data-testid="searchForm"
+                className="keycloak__events_search__form"
+                onSubmit={handleSubmit(onSubmit)}
+                isHorizontal
+              >
+                <TextControl
+                  name="user"
+                  label={t("userId")}
+                  data-testid="userId-searchField"
+                  isDisabled={!!user}
```
Comment (antikalk) [inline js/apps/admin-ui/src/events/UserEvents.tsx:250]: Disable input if a user has been passed to the component.

File: js/apps/admin-ui/src/events/UserEvents.tsx [hunk 4]
```diff
@@ -0,0 +1,476 @@
...
+              <div className="keycloak__searchChips pf-v5-u-ml-md">
+                {Object.entries(activeFilters).map((filter) => {
+                  const [key, value] = filter as [
+                    keyof UserEventSearchForm,
+                    string | EventType[],
+                  ];
+
+                  const disableClose =
+                    (key === "user" && !!user) ||
+                    (key === "client" && !!client);
```
Comment (antikalk) [inline js/apps/admin-ui/src/events/UserEvents.tsx:391]: Disable close button for user and/or client `Chip` if a value has been passed to the component. This shall prevent the user from removing the filter on a user or client details screen.

File: js/apps/admin-ui/src/events/EventsSection.tsx [hunk 5]
```diff
@@ -450,56 +50,7 @@ export default function EventsSection() {
...
-                emptyState={
-                  <ListEmptyState
-                    message={t("emptyUserEvents")}
-                    instructions={t("emptyUserEventsInstructions")}
-                  />
-                }
-                isSearching={Object.keys(activeFilters).length > 0}
-              />
-            </div>
+            <UserEvents />
```
Comment (antikalk) [inline js/apps/admin-ui/src/events/EventsSection.tsx:53]: Moved to dedicated UserEvents component.

**Review by agagancarczyk — APPROVED** [review]

**Pull request merged**